File: Problem137_SingerNumberII.py
from typing import List

class Solution:
    def singleNumber(self, nums: List[int]) -> int:
        ones = 0
        twos = 0
        for x in nums:
            ones = (ones ^ x) & ~twos
            twos = (twos ^ x) & ~ones
        return ones
    
nums = [0,1,0,1,0,1,99]
solve = Solution()
print(solve.singleNumber(nums))

File: Problem137_SingerNumberII.cpp
#include <bits/stdc++.h>
using namespace std;

class Solution {
public:
    int singleNumber(vector<int>& nums) {
        int ones=0, twos=0;
        for (int x : nums) {
            ones = (ones ^ x) & ~twos;
            twos = (twos ^ x) & ~ones;
        }
        return ones;
    }
};

int main() {
    vector<int> nums = {0,1,0,1,0,1,99};
    Solution solve;
    cout << solve.singleNumber(nums);
}